src/dune/file_tree.ml

```diff
@@ -232,6 +232,12 @@ end = struct
     ; recognize_jbuilder_projects : bool
     }
 
+  let equal x y =
```

Comment:
This seems very brittle to addition of new fields. How about something like this? (you lose symmetry a bit, but that seems all right)
```
  let equal { root; ancestor_vcs; recognize_jbuilder_projects } y =
    Path.Source.equal root y.root
    && Option.equal Vcs.equal ancestor_vcs y.ancestor_vcs
    && Bool.equal recognize_jbuilder_projects y.recognize_jbuilder_projects
```

(and good thing I tried to write this because only then I noticed `Vcs.equal x.ancestor_vcs x.ancestor_vcs` (x occurs on both sides))

Reply:
Good catch. I can't wait to address all this boilerplate for good.